src/Microsoft.Health.Api/Modules/InitializationModule.cs

```diff
@@ -38,17 +36,6 @@ public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
 
         private static void Configure(IApplicationBuilder app)
         {
-            ILogger logger = app.ApplicationServices.GetRequiredService<ILoggerFactory>().CreateLogger<InitializationModule>();
-
-            // start IStartable services.
-            foreach (var startable in app.ApplicationServices.GetService<IEnumerable<IStartable>>())
-            {
-                using (logger.BeginTimedScope($"Initializing {startable.GetType().Name}."))
-                {
-                    startable.Start();
```

Comment:
This PR is pretty scary for me. IStartable has a guarantee that it will happen after the construction of the IoC container, but before app initialization is complete. This PR changes that behavior to async background tasks that complete at unknown intervals. 

I think this behavior is also assumed by components that implement `IRequireInitializationOnFirstRequest`, where they expect the Start() to have completed, here that is no longer guaranteed.

Reply:
`StartAsync`s are awaited before the Host is considered "started" 
https://github.com/dotnet/extensions/blob/1264b9e2567027d2243f77f48cc08fdd576e3005/src/Hosting/Hosting/src/Internal/Host.cs#L47-L51

I don't think it's all that different from what we had before.

Reply:
I put a 30s delay in `StartAsync` and the there is no response to the web request. So I think we should be ok.

Reply:
Thanks for verifying John!

Reply:
It's actually simpler than that: they add the `GenericWebHostService` last so all other services are started first:
![image](https://user-images.githubusercontent.com/339432/96874343-03433580-1444-11eb-8fe9-f06b31f188c4.png)